### PR TITLE
[@types/urijs] Extended types for URI.buildQuery, URI.parseQuery, search and query

### PR DESCRIPTION
**Pull request by apepper**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

### CODE REVIEW

**Comment by danger-public** [comment]
Inspecting the JavaScript source for this package found some properties that are not in the .d.ts files.
The check for missing properties isn't always right, so take this list as advice, not a requirement.
## urijs ([<kbd>unpkg</kbd>](https://unpkg.com/browse/urijs@latest/))
was missing the following properties:
1. version
1. protocol_expression
1. idn_expression
1. punycode_expression
1. ip4_expression

<details>
<summary>as well as these 19 other properties...</summary>
<p>ip6_expression, find_uri_expression, findUri, defaultPorts, hostProtocols, invalid_hostname_characters, domAttributes, getDomAttribute, characters, decodePath, decodeUrnPath, recodePath, recodeUrnPath, buildQueryParameter, setQuery, hasQuery, ensureValidHostname, ensureValidPort, noConflict</p>
</details>

<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against 4b10fee602691db7bf7f46e238173b13abce42b0
</p>

**Comment by XVincentX** [comment]
I'm going to try to have a look at this on Monday!

**Review by marcelltoth — APPROVED** [review]
Looks good to me.

One note however: 
As you might know, tests are not actually ran here, just type-checked. This makes much of the tests we have (including ones you added) redundant. It's OK to have them I guess, but we could clean it up and make it much simpler should we decide to.

**Comment by marcelltoth** [comment]
@XVincentX oh you beat me by 17 seconds 😄 

@apepper Please wait for @XVincentX's review on Monday before merging this. Thank you!

File: types/urijs/test/urijs-dom.test.ts
Comment (XVincentX) [inline types/urijs/test/urijs-dom.test.ts:236]: I do not understand what are these tests for…are we testing the library in this repository? It does not really make much sense IMO, these files should be eliminated.

**Review by XVincentX — APPROVED** [review]
Nice stuff. From what I saw I suggest we completely remove the tests files; libraries do not get tested here.

**Comment by XVincentX** [comment]
>  It's OK to have them I guess, but we could clean it up and make it much simpler should we decide to.

Ok I saw your comment now. I agree with the suggestions, let's get rid of these files.

File: types/urijs/test/urijs-dom.test.ts
Comment (TeamworkGuy2) [reply types/urijs/test/urijs-dom.test.ts:236]: DefinitelyTyped tests are meant to check whether a type definition works correctly in common use cases.

**Review by TeamworkGuy2 — APPROVED** [review]

**Comment by apepper** [comment]
Since I don't know how you want to structure the tests in the future I'll simply merge this PR and leave it up to you to delete unwanted tests.

**Comment by apepper** [comment]
Ready to merge

**Comment by apepper** [comment]
When will this change be released on npm?

**Comment by marcelltoth** [comment]
> Since I don't know how you want to structure the tests in the future I'll simply merge this PR and leave it up to you to delete unwanted tests.

That's understandable as well, thanks for contributing.

> When will this change be released on npm?
It usually gets released in a few hours.

File: types/urijs/test/urijs-dom.test.ts
Comment (XVincentX) [reply types/urijs/test/urijs-dom.test.ts:236]: Fair enough, although to test types we should use the `// $ExpectType` assertions. In any case, it is not a big deal.

**Comment by apepper** [comment]
If I'm not mistaken the changes should now be live with version `1.19.10`. Thanks for all the reviews!

**Pull request merged**
